### PR TITLE
Moving history to new ~/.dlv config path

### LINKED ISSUE #81: Move readline history file to more sensible spot
Right now delve creates readline history file in cwd, but this is not acceptable going forward.

### PR DESCRIPTION
**Pull request by epipho**
Fixes #81 

This is also a setup for a ~/.dlv/,dlv.conf file to allow commands to be run at startup (i.e. change display modes).

### CODE REVIEW

File: client/cli/cli.go [hunk 1]
```diff
@@ -18,6 +20,7 @@ import (
 	"github.com/peterh/liner"
 )
 
+const configDir = ".dlv"
```
Comment (pnasrat) [inline client/cli/cli.go]: Probably want to think about win32 vs posix where you probably want different locations.

Comment (pnasrat) [inline client/cli/cli.go]: Actually I see you're using os/user to get homedir which I think will give USERPROFILE.  https://golang.org/src/os/user/lookup_windows.go#L115

**Comment by epipho** [comment]
It does give USERPROFILE on windows.  

While the official recommendation from MS is to use the LocalAppData system folder pretty much all the cross platform apps on my Win7 box appear to use a dot directory in USERPROFILE including VirtualBox, Vagrant, Maven, Netbeans, Babun, Atom, and CMake.

**Comment by derekparker** [comment]
This looks good but I'm going to wait until #88 lands to pull this in.

**Comment by derekparker** [comment]
@epipho now that #88 has been merged, can you take a look at rebasing this and pushing back up? I'd be happy to pull this in after that.

**Comment by epipho** [comment]
Yep I'll take a look this weekend.

**Comment by epipho** [comment]
Closing this in favor of #124

**Pull request closed without merging**